metaci/testresults/models.py

```diff
@@ -541,11 +542,20 @@ def summarize_week(cls, date):
         deleted = obsolete_objects.delete()
         cls.logger.info("Deleted %s", deleted)
 
-        new_objects = [cls(**values) for values in method_contexts]
-        cls.logger.info("Creating %s", len(new_objects))
-        created = cls.objects.bulk_create(new_objects)
-        cls.logger.info("Created %s", len(created))
-        return created
+        # from https://stackoverflow.com/a/3226719/11151197
+        def split_seq(iterable, size):
+            it = iter(iterable)
+            item = list(itertools.islice(it, size))
+            while item:
+                yield item
+                item = list(itertools.islice(it, size))
+
+        for batch in split_seq(method_contexts, 5000):
+            new_objects = [cls(**values) for values in batch]
+            cls.logger.info("Creating %s for %s", len(new_objects), date)
+            created = cls.objects.bulk_create(new_objects)
+            cls.logger.info("Created %s for %s", len(created), date)
+        return
```

Comment:
Can you take out the no-op `return` and move `split_seq` to metaci.utils?

Reply:
Thanks for merging master. I made the changes you suggested.